tests/example_app/src/app.py

```diff
@@ -56,5 +61,38 @@ def reverse_string(
     return ReverseStringOutput(reversed_string=reversed_string)
 
 
+@app.on_poll()
+def on_poll(
+    params: OnPollParams, client: SOARClient, asset: Asset
+) -> Iterator[Union[Container, Artifact]]:
+    # Create container first for artifacts
+    yield Container(
+        name="Network Alerts",
+        description="Some network-related alerts",
+        severity="medium",
+    )
+
+    # Simulate collecting 2 network artifacts that will be put in the network alerts container
+    for i in range(2):
+        client.save_progress(f"Processing network artifact {i}")
+
+        artifact = Artifact(
+            name=f"Network Alert {i + 1}",
+            label="alert",
+            severity="medium",
+            source_data_identifier=asset.base_url,
+            type="network",
+            description=f"Example network alert {i + 1} from polling operation",
+            data={
+                "alert_id": f"testalert-{datetime.now(timezone.utc).strftime('%Y%m%d')}-{i + 1}",
+                "source_ip": f"10.0.0.{i + 1}",
+                "destination_ip": "192.168.0.1",
+                "protocol": "TCP",
+            },
+        )
```

Comment:
I know this is a contrived example, but it would be more realistic for the SDI to be the alert ID:
```suggestion
        alert_id = f"testalert-{datetime.now(timezone.utc).strftime('%Y%m%d')}-{i + 1}"
        artifact = Artifact(
            name=f"Network Alert {i + 1}",
            label="alert",
            severity="medium",
            source_data_identifier=alert_id,
            type="network",
            description=f"Example network alert {i + 1} from polling operation",
            data={
                "alert_id": alert_id,
                "source_ip": f"10.0.0.{i + 1}",
                "destination_ip": "192.168.0.1",
                "protocol": "TCP",
            },
        )
```

Reply:
Agreed think that does make more sense, updated